src/hotspot/cpu/x86/x86.ad

```diff
@@ -9127,6 +9215,112 @@ instruct vreverse_byte64_reg(vec dst, vec src, vec xtmp1, vec xtmp2, rRegI rtmp)
   ins_pipe( pipe_slow );
 %}
 
+// ---------------------------------- Vector Count Leading Zeros -----------------------------------
+
+instruct vcount_leading_zeros_IL_reg_evex(vec dst, vec src) %{
+  predicate(is_clz_non_subword_predicate_evex(Matcher::vector_element_basic_type(n->in(1)),
+                                              Matcher::vector_length_in_bytes(n->in(1))));
+  match(Set dst (CountLeadingZerosV src));
+  format %{ "vector_count_leading_zeros $dst, $src" %}
+  ins_encode %{
+     int vlen_enc = vector_length_encoding(this, $src);
+     BasicType bt = Matcher::vector_element_basic_type(this, $src);
+     BasicType rbt = Matcher::vector_element_basic_type(this);
+     __ vector_count_leading_zeros_evex(bt, $dst$$XMMRegister, $src$$XMMRegister, xnoreg,
+                                        xnoreg, xnoreg, k0, noreg, true, vlen_enc);
+     // TODO: Once auto-vectorizer supports ConvL2I operation, CountLeadingZerosV
+     // should be succeeded by its corresponding vector IR and following
+     // special handling should be removed.
+     if (rbt == T_INT && bt == T_LONG) {
+       __ evpmovqd($dst$$XMMRegister, $dst$$XMMRegister, vlen_enc);
+     }
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct vcount_leading_zeros_IL_reg_evex_masked(vec dst, vec src, kReg mask) %{
+  predicate(is_clz_non_subword_predicate_evex(Matcher::vector_element_basic_type(n->in(1)),
+                                              Matcher::vector_length_in_bytes(n->in(1))));
+  match(Set dst (CountLeadingZerosV src mask));
+  format %{ "vector_count_leading_zeros $dst, $src, $mask" %}
+  ins_encode %{
+    int vlen_enc = vector_length_encoding(this, $src);
+    BasicType bt = Matcher::vector_element_basic_type(this, $src);
+    __ evmovdquq($dst$$XMMRegister, $src$$XMMRegister, vlen_enc);
+    __ vector_count_leading_zeros_evex(bt, $dst$$XMMRegister, $src$$XMMRegister, xnoreg, xnoreg,
+                                       xnoreg, $mask$$KRegister, noreg, true, vlen_enc);
```

Comment:
For the PopCountVL and CountLeadingZerosV for long, if the result type is T_INT the evpmovqd instruction needs to be generated.

Reply:
Its only through vector API will land in this control path and in case of vector API result lane size matches the source lane size.